src/ingest/subscriptions/post.js

```diff
@@ -7,19 +7,43 @@
 
 // load node utils
 const moment = require('moment');
+const { v4: uuidv4 } = require('uuid');
 
 // load eventhub utils
 const datastore = require('../../utils/datastore');
 const pubsub = require('../../utils/pubsub');
 const response = require('../../utils/response');
 
+//TODO: check IDs in ARD Core-API instead of dump
+const coreApi = require('../../data/coreApi.json');
+
 module.exports = async (req, res) => {
 	try {
+		// generate subscription name
+		let subIdent = 'subscription';
+		let prefix = `${global.PREFIX}.${subIdent}.${global.STAGE}`;
+
+		// check existence of user organization
+		let organizationExists = coreApi.some((entry) => {
+			if (req.user.organization?.id == entry.institution.id) {
+				return true;
+			}
+		});
+
+		if (!organizationExists) {
+			let orgId = req.user.organization?.id;
+			let orgName = req.user.organization?.name;
+			// return 401 error
+			return response.badRequest(req, res, {
+				status: 401,
+				message: `New subscriptions are not allowed for user '${req.user.email}'`,
+				errors: `The organization '${orgId}' (${orgName}) wasn't found in ARD Core-API`,
+			});
+		}
+
 		// map inputs
 		let subscription = {
-			name:
-				global.STAGE_CONFIG.pubsubPrefix +
-				`${req.user.organization}-${moment().format('YYYY-DDDD--x')}`,
+			name: `${prefix}.${req.user.organization?.name}.${uuidv4()}`,
```

Comment:
Should we explicitly check the existence of `organization.name` to avoid `null` subscriptions if the user wasn't entered properly?

Reply:
solved with https://github.com/swrlab/ard-eventhub/pull/26/commits/11a5714ae813f2690b2162ac3232084759e0773d